lib/validate_url.rb

```diff
@@ -72,16 +72,19 @@ def blacklisted_domain?(hostname)
         (hostname =~ blacklisted_domains_regex).present?
       end
 
-      def self.local?(hostname)
-        ip = begin
-               Resolv.getaddress(hostname)
-             rescue Resolv::ResolvError
-               nil
-             end
+      def self.local?(uri)
+        addr = begin
+                 Addrinfo.getaddrinfo(uri.hostname, get_port(uri), nil, :STREAM).sample
+               rescue SocketError
```

Comment:
getaddrinfo returns a SocketError when the DNS resolution fails. Should we return true in this case?

Reply:
It would return a empty array if it doesnt resolve to anything. SocketError is when there is no DNS reachable I believe.